assemblers/org.osbuild.qemu

```diff
@@ -136,10 +136,23 @@ def main(tree, output_dir, options, loop_client):
     extra_args = []
 
     # raw and vdi don't suppport compression
-    if fmt not in ("raw", "vdi"):
+    if fmt not in ("raw", "vdi", "vpc"):
         extra_args.append("-c")
 
-    subprocess.run(["qemu-img", "convert", "-O", fmt, *extra_args, image, f"{output_dir}/{filename}"], check=True)
+    if fmt != "vpc":
+        subprocess.run(["qemu-img", "convert", "-O", fmt, *extra_args, image, f"{output_dir}/{filename}"], check=True)
+    else:
+        # Implementation of the resize algorithm from here:
+        # https://docs.microsoft.com/en-us/azure/virtual-machines/linux/create-upload-generic
+        mb = 1024*1024
+        rounded_size = int((size/mb + 1) * mb)
+        subprocess.run(["qemu-img", "resize", image, rounded_size], check=True)
+        subprocess.run(["qemu-img", "convert",
+                        "-f", "raw",
```

Comment:
Is this necessary? We don't do it in the other branch.

Reply:
Try to run qemu-img by hand. In my case it printed a warning a requested to pass in explicitly.

Reply:
Fair enough. Let's do it for all the cases though.